ouroboros-network-api/src/Ouroboros/Network/PeerSelection/LedgerPeers/Type.hs

```diff
@@ -15,17 +22,137 @@ module Ouroboros.Network.PeerSelection.LedgerPeers.Type
+  -- ^ Internal use for version 1, use pattern synonym for public API
+  deriving (Eq, Show)
+
+-- |Public API to access snapshot data. Currently access to only most recent version is available.
+-- Nonetheless, serialisation from the node into JSON is supported for older versions via internal
+-- api so that newer CLI can still support older node formats.
+--
+pattern LedgerPeerSnapshot :: (WithOrigin SlotNo, [(AccPoolStake, (PoolStake, NonEmpty RelayAccessPoint))])
+                           -> LedgerPeerSnapshot
+pattern LedgerPeerSnapshot payload <- LedgerPeerSnapshotV1 payload where
+  LedgerPeerSnapshot payload = LedgerPeerSnapshotV1 payload
+
+{-# COMPLETE LedgerPeerSnapshot #-}
+
+-- | In case the format changes in the future, this function provides a migration functionality
+-- when possible.
+--
+migrateLedgerPeerSnapshot :: LedgerPeerSnapshot
+                          -> Maybe (WithOrigin SlotNo, [(AccPoolStake, (PoolStake, NonEmpty RelayAccessPoint))])
+migrateLedgerPeerSnapshot (LedgerPeerSnapshotV1 lps) = Just lps
+
+instance ToJSON LedgerPeerSnapshot where
+  toJSON (LedgerPeerSnapshotV1 (slot, pools)) =
+    object [ "Version" .= (1 :: Int)
+           , "SlotNo" .= slot
+           , "BigLedgerPools" .= [ object [ "AccStake" .= show accStake
+                                          , "ReStake"  .= show stake
+                                          , "Relays"   .= relays]
+                                 | (AccPoolStake accStake, (PoolStake stake, relays)) <- pools]]
+
+instance FromJSON LedgerPeerSnapshot where
+  parseJSON = withObject "LedgerPeerSnapshot" $ \v -> do
+    vNum <- v .: "Version"
+    parsedSnapshot <-
+      case (vNum :: Int) of
+        1 -> do
+          slot <- v .: "SlotNo"
+          bigPools <- v .: "BigLedgerPools"
+          bigPools' <- (forM bigPools . withObject "BigLedgerPools" $ \poolV -> do
+            accStake <-
+              poolV .: "AccStake"
+              >>= (\case
+                    Just accStake -> pure accStake
+                    Nothing    ->
+                      fail "Invalid AccStake value")
+                  . readMaybe
+            reStake <-
+              poolV .: "ReStake"
+              >>= (\case
+                    Just reStake -> pure reStake
+                    Nothing    ->
+                      fail "Invalid ReStake value")
+                  . readMaybe
+            relays <- poolV .: "Relays"
+            return (accStake, (reStake, relays))) <?> Key "BigLedgerPools"
+
+          return $ LedgerPeerSnapshotV1 (slot, bigPools')
+        _ -> fail $ "Network.LedgerPeers.Type: parseJSON: failed to parse unsupported version " <> show vNum
+    case migrateLedgerPeerSnapshot parsedSnapshot of
+      Just payload -> return $ LedgerPeerSnapshot payload
+      Nothing      -> fail "Network.LedgerPeers.Type: parseJSON: failed to migrate big ledger peer snapshot"
```

Comment:
We also wouldn't need this migration function since the codecs would have to be backward compatible and indexed by `LedgerSnapshotVersion`